File: system-test/typescript/src/testkit/dsl/port/myShop/given/steps/given-country.ts
import type { GivenStage } from '../given-stage.js';
import type { WhenStage } from '../../when/when-stage.js';
import type { ThenStage } from '../../then/then-stage.js';

export interface GivenCountry {
  withCode(country: string): GivenCountry;
  withTaxRate(taxRate: string | number): GivenCountry;
  and(): GivenStage;
  when(): WhenStage;
  then(): ThenStage;
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/port/myshop/given/steps/GivenCountry.java
package com.mycompany.myshop.testkit.dsl.port.myshop.given.steps;

import com.mycompany.myshop.testkit.dsl.port.myshop.given.steps.base.GivenStep;

public interface GivenCountry extends GivenStep {
    GivenCountry withCode(String country);

    GivenCountry withTaxRate(String taxRate);

    GivenCountry withTaxRate(double taxRate);
}
